presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/JoinMatcher.java

```diff
@@ -37,12 +37,14 @@
     private final JoinNode.Type joinType;
     private final List<ExpectedValueProvider<JoinNode.EquiJoinClause>> equiCriteria;
     private final Optional<Expression> filter;
+    private final Optional<JoinNode.DistributionType> distributionType;
 
-    JoinMatcher(JoinNode.Type joinType, List<ExpectedValueProvider<JoinNode.EquiJoinClause>> equiCriteria, Optional<Expression> filter)
+    JoinMatcher(JoinNode.Type joinType, List<ExpectedValueProvider<JoinNode.EquiJoinClause>> equiCriteria, Optional<Expression> filter, Optional<JoinNode.DistributionType> distributionType)
     {
         this.joinType = requireNonNull(joinType, "joinType is null");
         this.equiCriteria = requireNonNull(equiCriteria, "equiCriteria is null");
         this.filter = requireNonNull(filter, "filter can not be null");
+        this.distributionType = requireNonNull(distributionType, "distribtuionType cannot be null");
```

Comment:
I wonder what would be a good way to do that, since distribution type is already an optional.  What if we don't check for matches when you specify Optional.empty().  It would mean that you couldn't check that distributionType is specifically unset, but that doesn't seem like such a useful test (though maybe it's not expected behavior).  What do you think?